contracts/src/v0.8/ccip/DonIdClaimer.sol

```diff
@@ -0,0 +1,92 @@
+/// manage DON IDs without conflict or accidental reuse.
+/// @dev The contract maintains its own internal counter for DON IDs and ensures
+/// the next available ID is claimed and tracked by the contract. The sync function
+/// allows for alignment with the CapabilitiesRegistry.
+contract DonIDClaimer is ITypeAndVersion {
+  error ZeroAddressNotAllowed();
+  error Unauthorized(address caller);
+
+  string public constant override typeAndVersion = "DonIdClaimer 1.0.0-dev";
+  /// @notice The next available DON ID that is claimed and incremented
+  uint32 private s_nextDONId;  
+
+  /// @notice The address of the CapabilitiesRegistry contract used to fetch the next DON ID
+  address private immutable i_capabilitiesRegistry;
+
+  /// @notice Mapping to track authorized deployed keys 
+  mapping(address => bool) private authorizedDeployer; 
+
+  /// @notice Initializes the contract with the CapabilitiesRegistry address
+  /// @param capabilitiesRegistry The address of the CapabilitiesRegistry contract
+  constructor(address capabilitiesRegistry) {
+    if (capabilitiesRegistry == address(0)) revert ZeroAddressNotAllowed();  
+    i_capabilitiesRegistry = capabilitiesRegistry;
+
+    // Initializing the deployer authorization (owner can be the initial deployer)
+    authorizedDeployer[msg.sender] = true;
+
+    // Sync the initial s_nextDONId from the CapabilitiesRegistry contract
+    s_nextDONId = ICapabilityRegistry(i_capabilitiesRegistry).getNextDONId();
+  }
+
+  /// @notice Modifier to check if the caller is an authorized deployer
+  modifier onlyAuthorizedDeployer() {
+    if (!authorizedDeployer[msg.sender]) {
+      revert Unauthorized(msg.sender); 
+    }
+    _;
+  }
+
+  /// @notice Claims the next available DON ID and increments the internal counter
+  /// @dev The function increments s_nextDONId after returning the current value
+  /// @return uint32 The DON ID that was claimed
+  function claimNextDONId() external onlyAuthorizedDeployer returns (uint32) {
+    return s_nextDONId++;
+  }
+
+  /// @notice Synchronizes the next donID with the CapabilitiesRegistry and applies an offset
+  /// @param offset The offset to adjust the donID (useful when certain DON IDs are dropped)
+  /// @dev This can be used to synchronize with the CapabilitiesRegistry after some actions have occurred
+  function syncNextDONIdWithOffset(uint32 offset) external onlyAuthorizedDeployer {
+      address capabilitiesRegistry = i_capabilitiesRegistry;
+      s_nextDONId = ICapabilityRegistry(capabilitiesRegistry).getNextDONId() + offset;
+  } 
+
+  /// @notice Sets authorization status for a deployer address
+  /// @param senderAddress The address to be added or removed as an authorized deployer
+  /// @param allowed Boolean indicating whether the address is authorized (true) or revoked (false)
+  /// @dev Can only be called by an existing authorized deployer
+  function setAuthorizedDeployer(address senderAddress, bool allowed) external onlyAuthorizedDeployer {
```

Comment:
Let's have this function be onlyOwner protected, this contract can inherit from OwnerIsCreator like other contracts: https://github.com/smartcontractkit/chainlink/blob/b4e8ef801d9c1f74b5cdcf00aa85b3d861510145/contracts/src/v0.8/keystone/CapabilitiesRegistry.sol#L7C34-L7C66

Reply:
authorized deployers should still only be set by a central authority